tests/test_version_range.py

```diff
@@ -209,6 +211,32 @@ def test_PypiVersionRange_raises_ivr_for_invalid_ranges(self):
         except InvalidVersionRange as ivre:
             assert str(ivre).startswith("Unsupported character")
 
+    def test_NpmVersionRange_from_native(self):
+        """
+        Tests for NpmVersionRange.from_native
+        """
+        # test for `^version` or `Compatible with version`
+        npm_range = "^1.2.9"
+        npm_range_str = "vers:npm/>=1.2.9|<2.0.0"
+        expected = (
+            VersionConstraint(comparator=">=", version=SemverVersion(string="1.2.9")),
+            VersionConstraint(comparator="<", version=SemverVersion(string="2.0.0")),
+        )
+        version_range = NpmVersionRange.from_native(npm_range)
+        assert version_range.constraints == expected
+        assert version_range.to_string() == npm_range_str
+
+        # test for `~version` or `Approximately equivalent to version`
+        npm_range = "~3.8.2"
+        npm_range_str = "vers:npm/>=3.8.2|<3.9.0"
+        expected = (
+            VersionConstraint(comparator=">=", version=SemverVersion(string="3.8.2")),
+            VersionConstraint(comparator="<", version=SemverVersion(string="3.9.0")),
+        )
+        version_range = NpmVersionRange.from_native(npm_range)
+        assert version_range.constraints == expected
+        assert version_range.to_string() == npm_range_str
```

Comment:
These are two tests clubbed into one. Please split them accordingly. Something like `test_NpmVersionRange_from_native_range_with_caret_operator` `test_NpmVersionRange_from_native_range_with_tilde_operator`
or even better
`test_NpmVersionRange_from_native_range_with_compatible_with_version_operator` and 
`test_NpmVersionRange_from_native_range_with_approximately_equal_to_operator`

Reply:
Yeah that makes sense.. :)